tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/GoLanguageService.cs

```diff
@@ -39,131 +35,76 @@ public GoLanguageService(
 
     public override async Task<PackageInfo> GetPackageInfo(string packagePath, CancellationToken ct = default)
     {
+        var repoRoot = gitHelper.DiscoverRepoRoot(packagePath);
+        var commonPS1 = Path.Join(repoRoot, "eng", "common", "scripts", "common.ps1");
+
+        // The powershell outputs this:
+        // {
+        //   "Name": "sdk/messaging/azservicebus",
+        //   "Version": "1.10.1-beta.1",
+        //   "DirectoryPath": "../sdk/messaging/azservicebus",
+        //   "ServiceDirectory": "messaging/azservicebus",
+        //   "ReadMePath": "../sdk/messaging/azservicebus/README.md",
+        //   "ChangeLogPath": "../sdk/messaging/azservicebus/CHANGELOG.md",
+        //   "SdkType": "client",
+        //   "IsNewSdk": true,
+        //   "ReleaseStatus": "Unreleased",
+        //   "IncludedForValidation": false,
+        //   "CIParameters": {
+        //     "CIMatrixConfigs": []
+        //   },
+        //   "VersionFile": "/home/ripark/src/az/sdk/messaging/azservicebus/internal/constants.go",
+        //   "ModuleName": "azservicebus"
+        // }
         logger.LogDebug("Resolving Go package info for path: {packagePath}", packagePath);
-        var (repoRoot, relativePath, fullPath) = Parse(gitHelper, packagePath);
-        var (packageName, packageVersion) = await TryGetPackageInfoAsync(fullPath, ct);
-        
-        if (packageName == null)
+        string[] args = ["-c", $". {commonPS1}; Get-GoModuleProperties('{packagePath}') | ConvertTo-Json"];
```

Comment:
I'm good with this for now but I do think we should eventually move the implementation from the script into the c# code and invert this dependency and have the script call the cli.

Reply:
@richardpark-msft I meant we could just set the `SdkType` for now based on the module name because it's the current request. For the other missing properties, we could port them later.